numpy/_core/multiarray.py

```diff
@@ -1350,7 +1350,7 @@ def shares_memory(a, b, max_work=None):
     .. warning::
 
        This function can be exponentially slow for some inputs, unless
-       `max_work` is set to a finite number or ``MAY_SHARE_BOUNDS``.
+       `max_work` is set to a finite number or 0.
```

Comment:
maybe "to zero or a positive integer"?  (We could even just say "unless `max_work` is passed" and let the parameter docs speak for it.)